domains/middleware/authorization.go

```diff
@@ -150,3 +163,35 @@ func (am *authorizationMiddleware) authorize(ctx context.Context, op svcutil.Ope
 
 	return nil
 }
+
+func (am *authorizationMiddleware) checkDomain(ctx context.Context, session authn.Session, id string) error {
```

Comment:
@felixgateru 
What do you think about moving this logic to `pkg/authz` ?

Reply:
This sounds good as it wil reduce code repetition.